app/src/main/java/com/jerboa/model/CommunityViewModel.kt

```diff
@@ -61,26 +59,20 @@ class CommunityViewModel(account: Account, communityArg: Either<CommunityId, Str
 
     var sortType by mutableStateOf(SortType.Active)
         private set
-    var page by mutableIntStateOf(1)
-        private set
+    private var pageCursor: PaginationCursor? by mutableStateOf(null)
+
+    private var communityId: Int? by mutableStateOf(null)
+    private var communityName: String? by mutableStateOf(null)
 
     fun updateSortType(sortType: SortType) {
         this.sortType = sortType
     }
 
-    fun resetPage() {
-        page = 1
-    }
-
-    fun nextPage() {
-        page += 1
-    }
-
-    fun prevPage() {
-        page -= 1
+    private fun resetPage() {
+        pageCursor = null
```

Comment:
How does this resetPage logic work? in Appending it resets the value regardless the last pageCursor. Does this reset and cursor state even have purpose? Would the reset not be better be to remove the oldRes itself?

Reply:
If pageCursor is null / missing, then it just fetches the first page. If you call `getPosts(page_cursor = ...`, then it fetches the next page of results.

If you trace down resetPage(), you'll see its called for things like changing the sort type, clicking refresh, etc.